### DISCUSSION TITLE
Clicking on push notification on android is not calling onNotificationOpenedApp or getInitialNotification

### DISCUSSION THREAD

**Discussion by Aayushi1301** [invertase/react-native-firebase#4561] [Q&A]
I am using
"@react-native-firebase/app": "^9.0.0",
"@react-native-firebase/messaging": "^8.0.1"
As mentioned in the documentation, https://rnfirebase.io/messaging/notifications#handling-interaction, firebase listeners onNotificationOpenedApp or getInitialNotification should be called respectively on clicking on push notification from background/quit state. This works on iOS but on android these listeners are not called. I am using notification+data payload. 

I have a "click-action" as a string in data part. Can it be the reason for interfering with the firebase listeners. The same payload with "click-action" on iOS calls onNotificationOpenedApp and getInitialNotification.

**Comment by mikehardy** [comment]
This sounds like it would be an issue, not a discussion, and we would need the information in AndroidManifest and package.json to check.

This is usually something to do with a mis-configured integration with splash screens, specifically https://github.com/crazycodeboy/react-native-splash-screen/issues/289#issuecomment-502406454 - for which I recommend dumping that module and using react-native-boot-splash as it doesn't suffer the Intent.getExtras passing problem that causes this on Android normally

**Reply by Aayushi1301** [reply]
I am not using react native splash screen.

**Reply by mikehardy** [reply]
That's an assertion without the information we request on our new Issue template. You may still have some other launch Activity chaining to a second Activity, that is not passing Intent extras.
